components/order-app/stores/order-utils.ts

```diff
@@ -20,13 +25,16 @@ export const distanceFromTMS = async (googlePlacesObj: google.maps.places.PlaceR
   await insertGoogleMapsScript();
 
   const TMSLocation = new google.maps.LatLng(28.539307, -81.286839);
-  const result = await getDistance(TMSLocation, googlePlacesObj.geometry.location);
-  const distanceInMiles = Number(result.rows[0].elements[0].distance.text.split(' ')[0]);
-  return distanceInMiles;
+  if (googlePlacesObj.geometry && googlePlacesObj.geometry.location) {
```

Comment:
We should throw error if these aren't available instead of returning zero

Reply:
You are right! Is this fine

```ts
export const distanceFromTMS = async (googlePlacesObj: google.maps.places.PlaceResult) => {
  await insertGoogleMapsScript();

  const TMSLocation = new google.maps.LatLng(28.539307, -81.286839);
  if (!googlePlacesObj.geometry?.location) {
    throw new Error('Location not found');
  }

  const result = await getDistance(TMSLocation, googlePlacesObj.geometry.location);
  const distanceInMiles = Number(result.rows[0].elements[0].distance.text.split(' ')[0]);
  return distanceInMiles;
};
```
Lmk if you want me to change the error message.

Reply:
Yes